services/brig/test/integration/API/Search.hs

```diff
@@ -483,27 +481,16 @@ runBH opts =
 -- | This was copied from staging before the migration
 oldMapping :: Value
 oldMapping =
-  [aesonQQ| {
-      "user": {
-        "properties": {
-          "accent_id": {
-            "type": "byte",
-            "index": false
-          },
-          "handle": {
-            "type": "text"
-          },
-          "name": {
-            "type": "keyword",
-            "index": false
-          },
-          "normalized": {
-            "type": "text"
-          },
-          "team": {
-            "type": "keyword"
-          }
-        }
-      }
-    }
-  |]
+  Aeson.object
```

Comment:
I assume you've ruled out that `rm -rf ~/.stack && find / -name '*stack*' -exec rm -rf {} \;` would have helped?

Reply:
Yeah, I removed all of those files and rebuilt it, it still didn't work.

Reply:
1. What if you change it back now, after it works?  will it keep working?  :-)P
2. Did I see somebody mess with aeson versions on some branch?  Is that something worth trying?

Reply:
>     1. What if you change it back now, after it works?  will it keep working?  :-)P

Definitely worth trying, next compile is going to be this, hopefully this uncovers some mysteries.

>     2. Did I see somebody mess with aeson versions on some branch?  Is that something worth trying?

I tried several things (including compiling several times using `git bisect`) before I came to the conclusion that bumping aeson was a problem, somehow after taking back the aeson bump I could compile it. But it failed to compile from scratch. So, maybe some leftover state made it compile, I am not sure.

Reply:
When I replace the QQ expression with `undefined`, it compiles. Then if I add it back, it keeps compiling.

Reply:
Once it compiles, it keeps compiling even after `stack clean`. Only removing all the `.stack-work` directories make it fail again. I think this definitely needs to be reported to stack or ghc. @fisx Can you help me shrink it down so we don't have to make them compile.

Steps to reproduce this error:

```bash
docker run -it quay.io/akshay_wire/alpine-wire-builder bash
git clone https://github.com/wireapp/wire-server
cd wire-server
git checkout akshaymankar/cursed-alpine-qq-issue
stack build brig # See this fail
vi services/brig/test/integration/API/Search.hs # Replace the qq at the bottom with undefined
stack build brig # Passes
vi services/brig/test/integration/API/Search.hs # Add the qq back, or just git checkout -f
stack build brig # Passes - #TooCoolForReproducibility
```